lib/urlwatch/handler.py

```diff
@@ -127,11 +140,16 @@ def retrieve(self, timestamp=None, filter_func=None, headers=None,
             timestamp = email.utils.formatdate(timestamp)
             headers['If-Modified-Since'] = timestamp
 
-        if ' ' in self.location:
-            self.location, post_data = self.location.split(' ', 1)
+        if self.data is not None:
             log.info('Sending POST request to %s', self.location)
-        else:
-            post_data = None
+            # data might be dict or urlencoded string
+            if isinstance(self.data, dict):
+                # convert to urlencoded string
+                self.data = urlencode(self.data)
+            elif not isinstance(self.data, basestring):
+                # nuke / ignore other data (no string, no dict)
+                log.warning("invalid data parameter for url %s" % self.location)
```

Comment:
1. Use proper capitalization
2. For log calls, let the logger handle string formatting (`,` instead of `%`)
3. Also tell the user what the invalid parameter is and that we are ignoring it

Use:

    log.warning('Ignoring invalid data parameter for URL %s: %r', self.location, self.data)